controllers/goharbor/harborcluster/ctrl_setup.go

```diff
@@ -156,7 +155,7 @@ var harborClusterPredicateFuncs = predicate.Funcs{
 			return true
 		}
 		// when status was not changed and spec was not changed, not need reconcile
-		if equality.Semantic.DeepDerivative(oldObj.Spec, newObj.Spec) && oldObj.Status.Status == newObj.Status.Status {
+		if newObj.Status.ObservedGeneration == newObj.Generation {
```

Comment:
IMO the comment does not match the condition. The condition means _"Is the new object being reconciled?"_

In order to match the comment, I suggest:
```
If newObj.Generation == oldObj.Generation && equality.Semantic.DeepDerivative(oldObj.Statud, newObj.Status) {
```

WDYT?